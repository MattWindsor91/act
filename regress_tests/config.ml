(* The Automagic Compiler Tormentor

   Copyright (c) 2018--2019 Matt Windsor and contributors

   ACT itself is licensed under the MIT License. See the LICENSE file in the
   project root for more information.

   ACT is based in part on code from the Herdtools7 project
   (https://github.com/herd/herdtools7) : see the LICENSE.herd file in the
   project root for more information. *)

(** Tests the parsing of valid config files. *)

open Core

let pp_ast : Act_config.Ast.t Fmt.t =
  Fmt.using Act_config.Ast.sexp_of_t Sexp.pp_hum

let parse_config_failures ~(file : Fpath.t) ~(path : Fpath.t) :
    unit Or_error.t =
  ignore file ;
  let r = Act_config.Frontend.load (Plumbing.Input.of_fpath path) in
  Fmt.(
    pr "@[%a@]@."
      (result ~ok:pp_ast ~error:(any "UNEXPECTED ERROR:@ " ++ Error.pp)))
    r ;
  Result.ok_unit

let run (test_dir : Fpath.t) : unit Or_error.t =
  let full_dir = Fpath.(test_dir / "config" / "valid" / "") in
  Common.regress_on_files "Config parser (valid)" ~dir:full_dir ~ext:"conf"
    ~f:parse_config_failures

let command =
  Common.make_regress_command ~summary:"runs config parser regressions" run

let () = Command.run command
