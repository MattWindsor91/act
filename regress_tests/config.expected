# Config parser (valid) tests

## comments_only.conf

```
default {
  
}

fuzz {
  
}
```
## defaults.conf

```
default {
  try machine localhost
  try backend herd
}

fuzz {
  
}
```
## empty.conf

```
default {
  
}

fuzz {
  
}
```
## params.conf

```
default {
  
}

fuzz {
  action var.make 10
  set param action.cap.upper to 1000
  set flag action.enable to 2:1
  set flag action.pick-extra to no
  set flag mem.unsafe-weaken-orders to yes
}
```

Ran 4 test(s).
